### PR TITLE
Updated dirs.proj to build target only the core library.

### PR DESCRIPTION
**Pull request by dnduffy**
Updated dirs.proj to build target only the core library.
Added DownloadFile and ExtractZipArchive tasks.
Added FilesToSign and Signing.props and output messages.
Added Clean, CheckBuildParameters, DownloadCLI, AfterBuild, and PackageNuGet Targets .
Added dotnet CLI zip file download and extract and switched to dotnet build from msbuild.

### CODE REVIEW

File: src/Microsoft.ApplicationInsights.AspNetCore/project.json [hunk 1]
```diff
@@ -19,15 +19,16 @@
...
-  },
+    "dependencies": {
+        "Microsoft.AspNetCore.Hosting.Abstractions": "1.0.0-rc2-final",
+        "Microsoft.AspNetCore.Http.Abstractions": "1.0.0-rc2-final",
+        "Microsoft.AspNetCore.Mvc.ViewFeatures": "1.0.0-rc2-final",
+        "Microsoft.Extensions.Logging.Abstractions": "1.0.0-rc2-final",
+        "Microsoft.Extensions.Configuration": "1.0.0-rc2-final",
+        "Microsoft.Extensions.DiagnosticAdapter": "1.0.0-rc2-final",
+        "Microsoft.ApplicationInsights": "2.1.0-beta4",
+        "MicroBuild.Core": "0.2.0"
```
Comment (kartang) [inline src/Microsoft.ApplicationInsights.AspNetCore/project.json]: Please verify if removing this signs the package as it worked before. Thanks!

**Comment by kartang** [comment]
Fix the dependency to microbuild and :shipit:

File: Signing.props [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    <PropertyGroup>
+        <!-- TODO: Figure out how to get signing to work in MicroBuildV2 for .xproj without using this "unsupported" manual approach. -->
+        <PathToSigningDll>$(BUILD_ARTIFACTSTAGINGDIRECTORY)\MicroBuild\Plugins\MicroBuild.Plugins.Signing.1.0.200\build\MicroBuild.Signing.dll</PathToSigningDll>
```
Comment (kartang) [inline Signing.props]: Just curious: Is this the v1 microbuild?

Comment (dnduffy) [inline Signing.props]: No but it is doing things the V1 way that is "unsupported" now, however it is the only way I can get it to work at the moment.

Comment (kartang) [inline Signing.props]: Makes sense... :+1:

**Pull request merged**